File: Java/백준/Bronze/2884. 알람 시계/알람 시계.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;

public class Main{
    public static void main(String[] args) throws IOException{
        InputStream in = System.in;
        InputStreamReader reader = new InputStreamReader(in);
        BufferedReader br = new BufferedReader(reader);

        String inp[] = br.readLine().split(" ");
        
        int H = Integer.parseInt(inp[0]);
        int M = Integer.parseInt(inp[1]);

        H += 24;

        if (M < 45) {
            H -= 1;
            M += 15;
        } else {
            M -= 45;
        }

        H = H % 24;
        System.out.print(H);
        System.out.print(" ");
        System.out.println(M);


    }
}

File: Python/백준/Bronze/2884. 알람 시계/알람 시계.py
import sys
H, M = map(int, sys.stdin.readline().split())
M2 = M + H * 60 - 45
if M2 < 0:
    M2 = 24 * 60 + M2
H2 = M2 // 60
M2 = M2 % 60
print(H2, M2)